tmt/steps/prepare/distgit.py

```diff
@@ -246,6 +247,31 @@ def go(
         except tmt.utils.RunError as error:
             raise tmt.utils.PrepareError("Unable to 'rpmbuild -bp'.", causes=[error])
 
+        # Workaround around new rpm behavior, https://github.com/teemtee/tmt/issues/2987
+        # No hardcoded name, should keep working in the future
+        cmd = Command(
+            "rpmbuild",
+            "-bc",
+            "--short-circuit",
+            "--nodeps",
+            "--define",
+            '__spec_build_pre echo XYZbuilddir=%{_builddir}; exit 0',
+            spec_name,
+            *dir_defines)
+        outcome = guest.execute(command=cmd, cwd=source_dir).stdout or ''
+        match = re.search(r'XYZbuilddir=(.+)', outcome)
```

Comment:
Would it make sense to use some tmt-ish prefix? `XYZ` does not help much when one looks at logs and investigates issues, but seeing something like `tmt-prepare-disgit-builddir`, maybe even with run ID included, would help me understand who's responsible. I suppose it's not really a problem when looking at `log.txt`, but there might be some observers like systemtap or strace and other tools, watching things happening in the system, XYZ would be very anonymous.

Reply:
I went with `tmt-get-builddir`. IMO no point to make this too unique

Reply:
1b30375e